pySDC/projects/DAE/sweepers/RungeKuttaDAE.py

```diff
@@ -0,0 +1,181 @@
+from pySDC.implementations.sweeper_classes.Runge_Kutta import (
+    RungeKutta,
+    BackwardEuler,
+    CrankNicholson,
+    EDIRK4,
+    DIRK43_2,
+)
+
+
+class RungeKuttaDAE(RungeKutta):
+    r"""
+    Custom sweeper class to implement Runge-Kutta (RK) methods for general differential-algebraic equations (DAEs)
+    of the form
+
+    .. math::
+        0 = F(u, u', t).
+    
+    RK methods for general DAEs have the form
+
+    .. math::
+        0 = F(u_0 + \Delta t \sum_{j=1}^M a_{i,j} U_j, U_m),
+
+    .. math::
+        u_M = u_0 + \Delta t \sum_{j=1}^M b_j U_j.
+
+    In pySDC, RK methods are implemented in the way that the coefficient matrix :math:`A` in the Butcher
+    tableau is a lower triangular matrix so that the stages are updated node-by-node. This class therefore only supports
+    RK methods with lower triangular matrices in the tableau.
+
+    Parameters
+    ----------
+    params : dict
+        Parameters passed to the sweeper.
+
+    Attributes
+    ----------
+    du_init : dtype_f
+        Stores the initial condition for each step.
+
+    Note
+    ----
+    When using a RK sweeper to simulate a problem make sure the DAE problem class has a ``du_exact`` method since RK methods need an initial
+    condition for :math:`u'(t_0)` as well.
+
+    In order to implement a new RK method for DAEs a new tableau can be added in ``pySDC.implementations.sweeper_classes.Runge_Kutta.py``.
+    For example, a new method called ``newRungeKuttaMethod`` with nodes :math:`c=(c_1, c_2, c_3)`, weights :math:`b=(b_1, b_2, b_3)` and
+    coefficient matrix
+
+    ..math::
+        \begin{eqnarray}
+            A = \begin{pmatrix}
+                a_{11} & 0 & 0 \\
+                a_{21} & a_{22} & 0 \\
+                a_{31} & a_{32} & & 0 \\
+            \end{pmatrix}
+        \end{eqnarray}
+
+    can be implemented as follows:
+
+    >>> class newRungeKuttaMethod(RungeKutta):
+    >>>     nodes = np.array([c1, c2, c3])
+    >>>     weights = np.array([b1, b2, b3])
+    >>>     matrix = np.zeros((3, 3))
+    >>>     matrix[0, 0] = a11
+    >>>     matrix[1, :2] = [a21, a22]
+    >>>     matrix[2, :] = [a31, a32, a33]
+    >>>     ButcherTableauClass = ButcherTableau
+
+    The new class ``newRungeKuttaMethodDAE`` can then be used by defining the DAE class inheriting from both, this base class and class containing
+    the Butcher tableau:
+
+    >>> class newRungeKuttaMethodDAE(RungeKuttaDAE, newRungeKuttaMethod):
+    >>>     pass
+
+    More details can be found [here](https://github.com/Parallel-in-Time/pySDC/blob/master/pySDC/implementations/sweeper_classes/Runge_Kutta.py).
+    """
+
+    def __init__(self, params):
+        super().__init__(params)
+        self.du_init = None
+
+    def predict(self):
```

Comment:
You should raise `NotImplementedError` when someone tries to predict with something other than zero here.

Reply:
Yes, I agree!